zrml/neo-swaps/src/benchmarking.rs

```diff
@@ -97,13 +226,50 @@ fn create_market_and_deploy_pool<T: Config>(
     market_id
 }
 
+fn deposit_fees<T>(market_id: MarketIdOf<T>, amount: BalanceOf<T>)
+where
+    T: Config,
+{
+    let mut pool = Pools::<T>::get(market_id).unwrap();
+    assert_ok!(T::MultiCurrency::deposit(pool.collateral, &pool.account_id, amount));
+    assert_ok!(pool.liquidity_shares_manager.deposit_fees(amount));
+    Pools::<T>::insert(market_id, pool);
+}
+
+// Let `caller` join the pool of `market_id` after adding the  required funds to their account.
+fn add_liquidity_provider_to_market<T>(market_id: MarketIdOf<T>, caller: AccountIdOf<T>)
+where
+    T: Config,
+{
+    let pool = Pools::<T>::get(market_id).unwrap();
+    // Buy a little more to account for rounding.
+    let pool_shares_amount =
+        pool.liquidity_shares_manager.calculate_min_pool_shares_amount() + _1.saturated_into();
+    let ratio =
+        pool_shares_amount.bdiv(pool.liquidity_shares_manager.total_shares().unwrap()).unwrap();
+    let complete_set_amount =
+        pool.reserves.values().max().unwrap().bmul_ceil(ratio).unwrap() * 2u8.into();
+    assert_ok!(T::MultiCurrency::deposit(pool.collateral, &caller, complete_set_amount));
+    assert_ok_with_transaction!(T::CompleteSetOperations::buy_complete_set(
+        caller.clone(),
+        market_id,
+        complete_set_amount,
+    ));
```

Comment:
e7a431c6520ee860b9289c0b946974040b456a5d

Reply:
mhh.. seems not to be ideal, but okay for now.

I am questioning if we even should use `require_transactional` then at all. Because, if you have extrinsics that are already transactional and call a non-transactional function it will be inherited.